src/VisionNode.cpp

```diff
@@ -355,90 +193,134 @@ int main(int argc, char **argv)
                          uint32_t level)> dynamic_reconfigure_settings_callback =
         [&](iarc7_vision::VisionNodeConfig &config, uint32_t) {
             getDynamicSettings(config,
-                               private_nh,
                                line_extractor_settings,
-                               optical_flow_estimator_settings,
-                               dynamic_reconfigure_called);
-            ROS_ASSERT(gridline_estimator.onSettingsChanged());
-            ROS_ASSERT(optical_flow_estimator.onSettingsChanged());
-        };
-    dynamic_reconfigure_server.setCallback(dynamic_reconfigure_settings_callback);
-    iarc7_vision::RoombaEstimator roomba_estimator;
+                               optical_flow_estimator_settings);
+            dynamic_reconfigure_called = true;
+
+            if (gridline_estimator != nullptr) {
+                ROS_ASSERT(gridline_estimator->onSettingsChanged());
+            }
 
-    // Check for images at 100 Hz
-    ros::Rate rate (100);
+            if (optical_flow_estimator != nullptr) {
+                ROS_ASSERT(optical_flow_estimator->onSettingsChanged());
+            }
+        };
+    dynamic_reconfigure_server.setCallback(
+            dynamic_reconfigure_settings_callback);
 
-    // Wait for time to begin
+    // Wait for time to begin and settings to be fetched
     while (ros::ok() &&
            (ros::Time::now() == ros::Time(0) ||
            !dynamic_reconfigure_called)) {
         // wait
         ros::spinOnce();
     }
 
+    // Create vision processing objects
+    iarc7_vision::RoombaEstimator roomba_estimator;
+    gridline_estimator.reset(new iarc7_vision::GridLineEstimator(
+            line_extractor_settings,
+            grid_estimator_settings,
+            grid_line_debug_settings,
+            expected_image_format));
+    optical_flow_estimator.reset(new iarc7_vision::OpticalFlowEstimator(
+            optical_flow_estimator_settings,
+            optical_flow_debug_settings,
+            expected_image_format));
+
+    // Queue and callback for collecting images
+    std::deque<sensor_msgs::Image::ConstPtr> message_queue;
+    std::function<void(const sensor_msgs::Image::ConstPtr&)> image_msg_handler =
+        [&](const sensor_msgs::Image::ConstPtr& message) {
+            message_queue.push_back(message);
+        };
+
+    image_transport::ImageTransport image_transporter{nh};
+    image_transport::Subscriber sub = image_transporter.subscribe(
+        "/bottom_image_raw/image_raw",
+        100,
+        image_msg_handler);
+
     // Load the parameters specific to the vision node
     double startup_timeout;
     ROS_ASSERT(private_nh.getParam("startup_timeout", startup_timeout));
 
     size_t message_queue_item_limit = ros_utils::ParamUtils::getParam<int>(
             private_nh, "message_queue_item_limit");
 
+    // Loop rate
+    ros::Rate rate (300);
```

Comment:
100 Hz iterate without sleeping.

Reply:
Fixed

Reply:
Fixed.  We can't actually iterate without sleeping because `ros::spinOnce` is called further down the call stack in these other functions, so the queue can grow in there

Reply:
Although I'm realizing if there's a new image in the queue we do want to process it right away.  Fixing that now